### PR TITLE
1stClassStructs: Replace GT_LDOBJ with GT_OBJ

### PR DESCRIPTION
**Pull request by CarolEidt**
In preparation for using block nodes in assignments, change GT_LDOBJ to GT_OBJ.
Also, eliminate gtFldTreeList, which was only being used in a transitory fashion for x87
codegen - instead create the nodes on the fly as needed for stack fp codegen.
Additional minor cleanup.

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL - Ran full DDR+diffs (0 diffs)
Biggest impact is to the x86 frankenjit, due to changing stackfp.cpp not to generate the field nodes onto the GT_LDOBJ, but instead to generate them just when they are needed in codegencommon.cpp.
Other changes are purely refactoring & cleanup.

File: src/jit/importer.cpp [hunk 1]
```diff
@@ -13721,7 +13734,7 @@ bool Compiler::impReturnInstruction(BasicBlock *block, int prefixFlags, OPCODE &
                     //
                     // inlinee:
                     //      ...
-                    //      ldobj      !!T                 // this gets bashed to a GT_LCL_FLD, type TYP_INT
+                    //      obj      !!T                 // this gets bashed to a GT_LCL_FLD, type TYP_INT
```
Comment (mikedn) [inline src/jit/importer.cpp]: This probably shouldn't have been changed, it is the IL `ldobj` instruction.

File: src/jit/gtstructs.h [hunk 2]
```diff
@@ -80,7 +80,7 @@ GTSTRUCT_1(ArrOffs     , GT_ARR_OFFSET)
 GTSTRUCT_1(ArrIndex    , GT_ARR_INDEX) 
 GTSTRUCT_1(RetExpr     , GT_RET_EXPR) 
 GTSTRUCT_1(Stmt        , GT_STMT) 
-GTSTRUCT_1(LdObj       , GT_LDOBJ)
+GTSTRUCT_1(Obj       , GT_OBJ)
```
Comment (mikedn) [inline src/jit/gtstructs.h]: Nit: misaligned comma

File: src/jit/morph.cpp [hunk 3]
```diff
@@ -3153,7 +3153,7 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 #endif
             else // struct type
             {
-                /* We handle two opcodes: GT_MKREFANY and GT_LDOBJ */
+                // We handle twp opcodes: GT_MKREFANY and GT_OBJ
```
Comment (mikedn) [inline src/jit/morph.cpp]: two was accidentally changed to twp

File: src/jit/valuenum.cpp [hunk 4]
```diff
@@ -3567,7 +3567,7 @@ static genTreeOps genTreeOpsIllegalAsVNFunc[] = {
     GT_COPYOBJ,  // A kind of assignment.
     GT_ADDR,
     GT_ARR_BOUNDS_CHECK,
-    GT_LDOBJ,    // Heap memory.
+    GT_OBJ,    // Heap memory.
```
Comment (mikedn) [inline src/jit/valuenum.cpp]: Is the "heap memory" comment valid/useful?

File: src/jit/codegenlegacy.cpp [hunk 5]
```diff
@@ -18612,7 +18610,7 @@ void CodeGen::SetupLateArgs(GenTreePtr call)
                 } while (needOverwriteRegSrc != overwriteRegSrc);
             }
 
-            if ((arg->gtOper == GT_LDOBJ )&& (promotedStructLocalVarDesc == NULL))
+            if ((arg->gtOper == GT_OBJ )&& (promotedStructLocalVarDesc == NULL))
```
Comment (mikedn) [inline src/jit/codegenlegacy.cpp]: Nit: you could fix ` )&&` spacing

File: src/jit/compiler.h [hunk 6]
```diff
@@ -6874,25 +6874,49 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
...
+        case TYP_USHORT:    return SIMDUShortHandle;
+        case TYP_UBYTE:     return SIMDUByteHandle;
+        case TYP_SHORT:     return SIMDShortHandle;
+        case TYP_BYTE:      return SIMDByteHandle;
+        case TYP_LONG:      return SIMDLongHandle;
+        case TYP_UINT:      return SIMDUIntHandle;
+        case TYP_ULONG:     return SIMDULongHandle;
+        default:            break;
+        }
+        assert(!"Didn't find a class handle for simdType");
```
Comment (mikedn) [inline src/jit/compiler.h]: Wouldn't be better to place `assert/return` in the default case of the switch given that all other cases end with a return?

File: src/jit/codegenlegacy.cpp [hunk 7]
```diff
@@ -10551,13 +10551,9 @@ void                CodeGen::genCodeForTreeSmpOp(GenTreePtr tree,
...
-        case GT_LDOBJ:
-            // We need to decide whether all GT_LDOBJ nodes should be eliminated in importation/morphing (by
-            // translation into copyblk or ldind), or whether it is legal to have as arguments to this
-            // method, where we have to generate code for them.
-            // On AMD64, at least, we can get here...
-            NYI("Handle GT_LDOBJ, or eliminate them earlier.");
-            unreached();
+        case GT_OBJ:
+            // All GT_OBJ nodes must have been morphed prior to this.
+            noway_assert("Should not see a GT_OBJ node during CodeGen.");
```
Comment (BruceForstall) [inline src/jit/codegenlegacy.cpp]: This assert will never fire. You need a !

**Comment by BruceForstall** [comment]
Except for one comment, LGTM

File: src/jit/importer.cpp (hunk 1 above)
Comment (CarolEidt) [inline src/jit/importer.cpp]: Indeed, thanks.

File: src/jit/valuenum.cpp (hunk 4 above)
Comment (CarolEidt) [inline src/jit/valuenum.cpp]: No, and never was I don't think.

Comment (CarolEidt) [inline src/jit/valuenum.cpp]: Actually, I think it should say "May reference heap memory."

File: src/jit/compiler.h (hunk 6 above)
Comment (CarolEidt) [inline src/jit/compiler.h]: I moved the assert there, but I'm not crazy about putting the return on the same line, so I left that after the switch.

**Comment by CarolEidt** [comment]
Made changes based on feedback.

File: src/jit/codegencommon.cpp [hunk 8]
```diff
@@ -842,7 +842,7 @@ void Compiler::compUpdateLifeVar(GenTreePtr tree, VARSET_TP* pLastUseVars)
 #if !defined(_TARGET_AMD64_) // no addr nodes on AMD and experimenting with with encountering vars in 'random' order
     if (ForCodeGen)
     {
-        if (tree->gtOper == GT_LDOBJ)
+        if (tree->gtOper == GT_OBJ)
         {
             // The tree must have the particular form LDOBJ(ADDR(LCL)); no need to do the check below.
```
Comment (briansull) [inline src/jit/codegencommon.cpp]: change comment to "...form OBJ(ADDR(LCL)"

File: src/jit/codegenxarch.cpp [hunk 9]
```diff
@@ -1522,7 +1522,7 @@ CodeGen::genStructReturn(GenTreePtr treeNode)
 
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
     noway_assert((op1->OperGet() == GT_LCL_VAR) ||
-                 (op1->OperGet() == GT_CALL));
+                    (op1->OperGet() == GT_CALL));
```
Comment (briansull) [inline src/jit/codegenxarch.cpp]: extra spaces here

**Comment by briansull** [comment]
LGTM, with minor comments

**Pull request merged**